app/models/energy/ScheduleFile.py

```diff
@@ -0,0 +1,59 @@
+"""Schedule File Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.ScheduleBase import ScheduleContinuous, ScheduleDiscrete, ScheduleNumericType, ScheduleUnitType
+
+
+class ScheduleFile(BaseModel):
+    """A list of values used to create a .csv file for the entire year or a period of
+  analysis within the year."""
+
+    type: Enum('ScheduleFile', {'type': 'ScheduleFile'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    start_day_of_year: int = Schema(
```

Comment:
@chriswmackey should I add the start_day_of_year as start_month and start_day the way it is described in OpenStudio?

Reply:
We usually think of start_day_of_year as the number of days from 1/1 (starting at one), so July 4th
could be day of year 185

https://www.esrl.noaa.gov/gmd/grad/neubrew/Calendar.jsp

Reply:
If the ScheduleFixedInterval supports it, this should be the `Date` object noted above rather than a `start_day_of_year`.  However, I realize that just the Date alone is not enough to know whether the values are for a leap year or not. So @macumber , does the `ScheduleInterval` have a property to denote whether it is for a leap year in the same way that `ScheduleFile` knows that it is a leap year because there are 8784 values in the fie?  Along these lines, it would be really helpful to see some sample code for the `ScheduleInterval`.  I have not yet been able to successfully implement one myself and I still don't quite understand how they are ultimately formatted into the IDF.

Reply:
It is a bit weird to work with but I can help with the OpenStudio pieces.  

They are currently translated to `Schedule:Compact`, https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateScheduleFixedInterval.cpp#L75.  However, this might change, we have thoughts of translating these to `Schedule:File` instead, https://github.com/NREL/OpenStudio/issues/50

Here is some example code using this schedule type from the radiance measure

        ```
originalLightsValues = lightsTimeseries.values
        lightsValues = OpenStudio::Vector.new(averageIlluminances.size)
        averageIlluminances.each_index do |i|
          dimmingResponse = [(daylightSetpoint - averageIlluminances[i]) / daylightSetpoint, 0].max
          lightsValues[i] = dimmingResponse * originalLightsValues[i]
        end

        # get max lighting power
        lightingLevel = OpenStudio.maximum(lightsValues)

        if lightingLevel <= 0.0
          print_statement("Thermal Zone '#{thermalZone.name}' lighting power is less than or equal to 0, skipping", runner)
          next
        end

        print_statement("Thermal Zone '#{thermalZone.name}' lighting power is: #{lightingLevel.round(0)} W", runner)

        # normalize lights values
        averageIlluminances.each_index do |i|
          lightsValues[i] = lightsValues[i] / lightingLevel
        end

        startDate = OpenStudio::Date.new(OpenStudio::MonthOfYear.new(1), 1)
        interval = OpenStudio::Time.new(0, 1, 0)
        timeseries = OpenStudio::TimeSeries.new(startDate, interval, lightsValues, 'W')
```

Reply:
Here is a script you can play with to look at the ScheduleFixedInterval class and translation to IDF.  Just save it run it with the CLI `openstudio test_schedule_file.rb`:

```
year = 2013
num_samples = 8760
#num_samples = 8784

values = OpenStudio::Vector.new(num_samples)
(0...num_samples).each do |i|
  values[i] = i % 24
end

startDate = OpenStudio::Date.new(OpenStudio::MonthOfYear.new(1), 1, year)
interval = OpenStudio::Time.new(0, 1, 0)
timeseries = OpenStudio::TimeSeries.new(startDate, interval, values, 'W')

model = OpenStudio::Model::Model.new
schedule = OpenStudio::Model::ScheduleInterval::fromTimeSeries(timeseries, model)

puts model

ft = OpenStudio::EnergyPlus::ForwardTranslator.new
idf = ft.translateModel(model)

puts idf

```

Reply:
Thanks for posting the sample code, @macumber .  I got the chance to run it and it has definitely given me some perspective.  I was initially thinking that the `FixedInterval` schedule was better suited to our needs but now I see it has many more issues than I realized.  The bug in the github issue is the major one that is the deal-breaker but, on top of that I see that just a single year-long compact schedule is so big that it can't be opened in IDFEditor.  Also, all of the extra comments and E+ tags means that the schedule file size is almost 100 times that of its CSV counterpart.  So, altogether, I am positive now that we want to be using ScheduleFile in the measure.  This brings up the question of whether we should stick with the naming convention of ScheduleFixedInterval in the schema and, after thinking about it, I feel that this is still appropriate since we ask the user for a list of values rather than a file.  So we can keep the name but, knowing that there will be using ScheduleFile on the measure end, I am ok if we wanted to stick with the convention of using the start_day_of_year instead of inputting a whole date object.  If there's a datetime module in Ruby, though, it would be nice if we can accept the date object and then convert it to a start_day_of_year in the measure.

On a side note, I had no idea that the year description had all of these capabilities.  The year description object isn't accessible from C# bindings (I have spent quite a few hours in the past trying to access it without any success).  For the sake of the legacy plugins, do you think that the ability to set the leap year will ever be exposed on the C# bindings?

Reply:
We have on our list that we should translate schedule interval to schedule file in e+, that is a bug/improvement we could make on the os side.  I think the year description should be available as a method in the Model class, I can try it out sometime when I’m at my computer

Reply:
Thanks for the quick response @macumber .  If the change over to ScheduleFile is something that you would be able to make within the IN2 timeline, then I agree that sounds like the best solution.  I know that we budgeted some hours for you to make OpenStudio edits in the IN2 budget and this purpose definitely seems worthy of using those hours.
Investigating the YearDescription in C# bindings is a lower priority since we have workarounds for it in the legacy plugins.  But it still would be really nice to get to at some point.

Reply:
I think I can do it for IN2, its good for everyone